test/unit/data/test_galaxy_mapping.py

```diff
@@ -384,6 +384,7 @@ def test_metadata_spec(self):
         d = self.model.HistoryDatasetAssociation(extension="interval", metadata=metadata, sa_session=self.model.session)
         assert d.metadata.chromCol == 1
         assert d.metadata.anyAttribute is None
+        assert d.metadata.items()
```

Comment:
This passes also without the changes in `lib/galaxy/model/metadata.py` , what error are you seeing?

Reply:
`'NoneType' object is not callable`

Reply:
https://sentry.galaxyproject.org/sentry/main/issues/1330503/
```
TypeError: 'NoneType' object is not callable
  File "galaxy/util/template.py", line 80, in fill_template
    return unicodify(t, log_exception=False)
  File "galaxy/util/__init__.py", line 1039, in unicodify
    value = str(value)
  File "Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1613369615_6865764_72689.py", line 157, in respond
TypeError: 'NoneType' object is not callable
  File "galaxy/jobs/runners/__init__.py", line 235, in prepare_job
    job_wrapper.prepare()
  File "galaxy/jobs/__init__.py", line 1126, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "galaxy/tools/evaluation.py", line 451, in build
    raise e
  File "galaxy/tools/evaluation.py", line 447, in build
    self.__build_config_files()
  File "galaxy/tools/evaluation.py", line 521, in __build_config_files
    self.__write_workdir_file(config_filename, config_text, param_dict, is_template=is_template)
  File "galaxy/tools/evaluation.py", line 615, in __write_workdir_file
    value = fill_template(content, context=context, python_template_version=self.tool.python_template_version)
  File "galaxy/util/template.py", line 124, in fill_template
    python_template_version=python_template_version,
  File "galaxy/util/template.py", line 126, in fill_template
    raise first_exception or e
  File "galaxy/util/template.py", line 80, in fill_template
    return unicodify(t, log_exception=False)
  File "galaxy/util/__init__.py", line 1039, in unicodify
    value = str(value)
  File "Cheetah/Template.py", line 1053, in __unicode__
    return getattr(self, mainMethName)()
  File "cheetah_DynamicallyCompiledCheetahTemplate_1613361333_4529872_28988.py", line 154, in respond
```

Reply:
The fix is live on usegalaxy.org, https://usegalaxy.org/u/marius/h/imported-smorg-1-3-mapping-workflow, items 14, 19 and 20 are without the fix, 21 is with. I can also reproduce it locally, but I agree the unit test passes without the change. I used it to check for possible recursion issues here.